packages/core/src/storages/request_queue_v2.ts

```diff
@@ -109,6 +109,11 @@ export class RequestQueue extends RequestProvider {
      * @inheritDoc
      */
     override async fetchNextRequest<T extends Dictionary = Dictionary>(): Promise<Request<T> | null> {
+
+        // TODO Kuba: We are missing there check for the queuePausedForMigration and for aborting.
+        // If we did not pause it here, we would be fetching and locking requests.
+        // The raise condition is here that we call _clearPossibleLocks(), but after that we can still fetch and lock requests, so the requests will be locked and never unlocked.
```

Comment:
Soo... we should just return `null` if `queuePausedForMigration || aborting`? That should prevent us from calling `listAndLockHead` again.

Reply:
I didn't go deep in the implementation of each scrapers, but overall once a migrating or aborting event happens, we should stop working with queue and persist.